src/client.rs

```diff
@@ -176,9 +167,11 @@ impl Tidal {
                 builder
             };
 
+            debug!("request builder: {:?}", builder);
             builder.send().await.map_err(ClientError::from)?
```

Comment:
Should `await` be used with the `?` operator?

Reply:
@jpalvarezl why not? `send()` returns a `Future<Option = Result<reqwest::Response, reqwest::Error>>` since it will work as kind of like promise and our function signature is `ClientResult<T>` aka. `Result<T, ClientError>` I don't see why not bubble up the errors after transforming them to a `ClientError` with `map_err(ClientError::from)`.

Could you elaborate more about your comment? I might be missing somehting

Reply:
I think Ive just seen this usually used as `àwait?` so I thought it was part of the syntax somehow. Maybe it's just to get the type of `Ok` directly as the type of the future instead... never mind.